runtime/onert/core/src/ir/train/UseDefGenerator.test.cc

```diff
@@ -0,0 +1,2027 @@
+
+    enableAllBackwarding(tgraph);
+
+    EXPECT_NO_THROW(tgraph.setTrainingUseDefs(train::UseDefGenerator{tgraph}()));
+    EXPECT_NO_THROW(tgraph.verify());
+
+    train::TrainingOperationIndex forwarding_fc_index{fc_index, true};
+    train::TrainingOperationIndex forwarding_ba_index{ba_index, true};
+    train::TrainingOperationIndex forwarding_loss_index{loss_index, true};
+    train::TrainingOperationIndex backwarding_fc_index{fc_index, false};
+    train::TrainingOperationIndex backwarding_ba_index{ba_index, false};
+    train::TrainingOperationIndex backwarding_loss_index{loss_index, false};
+
+    std::vector<train::TrainingOperationIndex> expected_forwarding_input_uses{forwarding_fc_index,
+                                                                              backwarding_fc_index};
+    std::vector<train::TrainingOperationIndex> expected_forwarding_input_def{};
+    const auto expected_forwarding_input = createUseDefChain(
+      tgraph.operands().at(input), expected_forwarding_input_uses, expected_forwarding_input_def);
+
+    std::vector<train::TrainingOperationIndex> expected_forwarding_weights_uses{
+      forwarding_fc_index, backwarding_fc_index};
+    std::vector<train::TrainingOperationIndex> expected_forwarding_weights_def{};
+    const auto expected_forwarding_weights =
+      createUseDefChain(tgraph.operands().at(weights), expected_forwarding_weights_uses,
+                        expected_forwarding_weights_def);
+
+    // forwarding output of FC is not used in backwarding if activation is none
+    std::vector<train::TrainingOperationIndex> expected_forwarding_ba_input1_uses{
+      forwarding_ba_index, backwarding_ba_index};
+    std::vector<train::TrainingOperationIndex> expected_forwarding_ba_input1_def{
+      forwarding_fc_index};
+    const auto expected_forwarding_ba_input1 =
+      createUseDefChain(tgraph.operands().at(ba_input1), expected_forwarding_ba_input1_uses,
+                        expected_forwarding_ba_input1_def);
+
+    std::vector<train::TrainingOperationIndex> expected_forwarding_ba_input2_uses{
+      forwarding_ba_index, backwarding_ba_index};
+    std::vector<train::TrainingOperationIndex> expected_forwarding_ba_input2_def{};
+    const auto expected_forwarding_ba_input2 =
+      createUseDefChain(tgraph.operands().at(ba_input2), expected_forwarding_ba_input2_uses,
+                        expected_forwarding_ba_input2_def);
+
+    std::vector<train::TrainingOperationIndex> expected_forwarding_y_pred_uses{
+      forwarding_loss_index, backwarding_loss_index};
+    std::vector<train::TrainingOperationIndex> expected_forwarding_y_pred_def{forwarding_ba_index};
+    const auto expected_forwarding_y_pred =
+      createUseDefChain(tgraph.operands().at(y_pred), expected_forwarding_y_pred_uses,
+                        expected_forwarding_y_pred_def);
+
+    std::vector<train::TrainingOperationIndex> expected_forwarding_y_true_uses{
+      forwarding_loss_index, backwarding_loss_index};
```

Comment:
This expected "uses" means forwarding "loss" node and backwarding "loss" node are added as "uses" of "y_true" input of forwarding "loss" node.